udata/core/dataset/rdf.py

```diff
@@ -570,36 +638,41 @@ def dataset_from_rdf(graph: Graph, dataset=None, node=None):
     if temporal_coverage:
         dataset.temporal_coverage = temporal_from_rdf(d.value(DCT.temporal))
 
-    # Adding some metadata to extras - may be moved to property if relevant
-    access_rights = rdf_value(d, DCT.accessRights)
-    if access_rights:
-        dataset.extras["harvest"] = {
-            "dct:accessRights": access_rights,
-            **dataset.extras.get("harvest", {}),
-        }
-    provenance = [p.value(RDFS.label) for p in d.objects(DCT.provenance)]
-    if provenance:
-        dataset.extras["harvest"] = {
-            "dct:provenance": provenance,
-            **dataset.extras.get("harvest", {}),
-        }
+    provenances = provenances_from_rdf(d)
+    if provenances:
+        add_dcat_extra(dataset, "provenance", provenances)
 
-    licenses = set()
+    resources_licenses_hints = set()
+    resources_access_rights = []
     for distrib in d.objects(DCAT.distribution | DCAT.distributions):
         resource_from_rdf(distrib, dataset)
-        for predicate in DCT.license, DCT.rights:
-            value = distrib.value(predicate)
-            if isinstance(value, (URIRef, Literal)):
-                licenses.add(value.toPython())
-            elif isinstance(value, RdfResource):
-                licenses.add(value.identifier.toPython())
+        resources_access_rights.append(access_rights_from_rdf(distrib))
+        # include both dct:license and dct:rights as licenses hints from resources
+        resources_licenses_hints |= licenses_from_rdf(distrib)
+        resources_licenses_hints |= rights_from_rdf(distrib)
 
     for additionnal in d.objects(DCT.hasPart):
         resource_from_rdf(additionnal, dataset, is_additionnal=True)
 
+    dataset_access_rights = access_rights_from_rdf(d)
+    if not dataset_access_rights and resources_access_rights:
+        dataset_access_rights = infer_dataset_access_rights(resources_access_rights)
```

Comment:
We could have
```
dataset_access_rights = infer_dataset_access_rights(dataset_access_rights, resources_access_rights)
```
with the fallback logic on resources access rights inside the function, as we have for `License.guess`?

Reply:
https://github.com/opendatateam/udata/pull/3054/commits/61548df9c9aba23919e14dc87beb229ce8513cf0